### PR TITLE
fix: check preconditions on correct state

### PR DESCRIPTION
**Pull request by thomaseizinger**
When shrinking a state machine test, proptest either shrinks a transition, deletes a transition or attempts to shrink the initial state. The `check_acceptable` function however uses the `last_valid_initial_state` which isn't the correct one if we just shrank / complicated the initial state.

### CODE REVIEW

File: proptest-state-machine/src/strategy.rs [hunk 1]
```diff
@@ -606,7 +617,7 @@ impl<
             }
             Some(InitialState) => {
                 if self.initial_state.complicate()
-                    && self.check_acceptable(None)
```
Comment (thomaseizinger) [inline proptest-state-machine/src/strategy.rs:609]: This is the bug (in my opinion). Previously, we checked whether the **last** valid state was acceptable but when we are shrinking the `initial_state`, we need to check whether the current one is acceptable.

File: proptest/src/test_runner/runner.rs
Comment (thomaseizinger) [inline proptest/src/test_runner/runner.rs:1]: Happy to split these out into a separate PR if you'd like. I found them helpful during debugging.

**Comment by thomaseizinger** [comment]
@matthew-russo Hopefully you don't mind the ping! Would appreciate if you could have a look at this and let me know your opinion on it! :)

**Review by matthew-russo — APPROVED** [review]
Seems okay to me but I've added Tomáš as well to double check

**Comment by tzemanovic** [comment]
Nice catch, thank you!

**Review by tzemanovic — APPROVED** [review]

**Comment by nickelization** [comment]
I independently ran into this same bug the other day, and once I figured out it was a proptest bug I was happy to see it's already been fixed 😄 Curious if there are plans to cut a new release anytime soon, I'd love to have this fix in a published version of the crate.

**Comment by tzemanovic** [comment]
> I independently ran into this same bug the other day, and once I figured out it was a proptest bug I was happy to see it's already been fixed 😄 Curious if there are plans to cut a new release anytime soon, I'd love to have this fix in a published version of the crate.

apologies for the long delay, will prepare a patch release today

**Comment by tzemanovic** [comment]
released in v0.3.1
